### PR TITLE
[#2] [Infrastructure] Setup CI with Github Actions for code linting

### LINKED ISSUE #2: [Infrastructure] Setup CI with Github Actions for code linting
## Why

Code linting helps to enforce a consistent style in our codebase and can identify potential issues before they become bugs.

Automating code linting as part of our CI process ensures that code quality is maintained and enforced at all times, rather than relying on manual code reviews to catch potential issues.

Setting up code linting in our CI process helps to improve the stability of our codebase, as it can catch issues that might otherwise go unnoticed.

## Acceptance Criteria

 - Code linting is set up as a step in our CI process on Github Actions.
 - Linting is triggered on every push to the repository and on every pull request.
 - The linting step is configured to fail the build if linting errors are detected.
 - The configuration for the linting step is stored in the repository, so that it is easy to maintain and update.
 - Linters to enable: only terraform fmt for now

### PR DESCRIPTION
**Pull request by Nihisil**
- Close #2

## What happened 👀

`terraform fmt` and `tfsec` linters were set up for the project to be run on each push to the repository.

## Proof Of Work 📹

For the test I added not formated code to the repo, CI was failed:

image

Otherwise CI is green:

image

### CODE REVIEW

**Review by andyduong1920 — APPROVED** [review]

File: .github/workflows/lint_code.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+jobs:
+  linting:
+    name: Lint Terraform files
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Cancel previous runs
+        uses: styfle/cancel-workflow-action@0.10.0
```
Comment (longnd) [inline .github/workflows/lint_code.yml]: there is a newer version of the action, you can consider using it
```suggestion
        uses: styfle/cancel-workflow-action@0.11.0
```

File: .github/workflows/lint_code.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        uses: hashicorp/setup-terraform@v2
+        with:
+          terraform_version: ${{ env.TERRAFORM_VERSION }}
+
+      - name: Run Terraform format
+        run: terraform fmt -recursive -check
+
+      - name: Run tfsec linter
+        id: tfsec
+        uses: aquasecurity/tfsec-action@v1.0.2
```
Comment (longnd) [inline .github/workflows/lint_code.yml:36]: Same here, this version is outdated
```suggestion
        uses: aquasecurity/tfsec-action@v1.28.0
```

**Review by longnd — APPROVED** [review]
some minor things only :)

File: .github/workflows/lint_code.yml (hunk 1 above)
Comment (Nihisil) [reply .github/workflows/lint_code.yml]: Thanks. This one was upgraded here: https://github.com/Nihisil/nimble-devops-ic-infrastructure/pull/18/commits/72364cbdceb6fb5034419ae01da350d69294b53b

File: .github/workflows/lint_code.yml (hunk 2 above)
Comment (Nihisil) [reply .github/workflows/lint_code.yml:36]: Can you please share where did you find new version?

Here latest version is `1.0.2` :thinking: 
- https://github.com/marketplace/actions/tfsec-action
- https://github.com/aquasecurity/tfsec-action

Comment (Nihisil) [reply .github/workflows/lint_code.yml:36]: `1.28.0` is tfsec version, but GH action doesn't have new version. I think all good here :slightly_smiling_face:

**Pull request merged**